src/bin/connect.ml

```diff
@@ -71,11 +71,14 @@ module Hvsock = struct
     failwith "Hyper-V socket forwarding not initialised"
   | Some sockaddr ->
     let description = "hvsock" in
-    Host.Sockets.register_connection description >>= fun idx ->
-    let fd = F.Socket.create () in
-    F.Socket.connect fd sockaddr >|= fun () ->
-    let flow = F.connect fd in
-    { idx; flow }
+    begin match Connection_limit.register description with
+    | Error (`Msg m) -> Lwt.fail_with m
+    | Ok idx ->
```

Comment:
This looks a bit like `Lwt_result.get_exn` but specialised for the Msg case -- https://ocsigen.org/lwt/latest/api/Lwt_result

Perhaps define a quick operator for that and use it in all the places you do the Error -> Lwt.fail_with conversion?

Reply:
I added a
```ocaml
let ( >>*= ) m f = m >>= function
  | Error (`Msg m) -> Lwt.fail_with m
  | Ok x -> f x
```
and used it in a bunch of places.